### PR TITLE
New Construct & Animations

### LINKED ISSUE #162: Direct computation of evolution timeseries
Doing `construct(evolve(p, bd, t)...)` has the disadvantage that it allocates a lot of unecessary stuff inbetween. It is better to define one extra new function that directly outputs the timeseries of positions and velocities and time.

To avoid code repetition, it is best to define also an internal function `extrapolate(p, prevpos, t)` that given the particle, the previous position and the time to go to the new position (which is `p.pos`), it extrapolates in between the position and velocity timeseries (notice that for `Particle` this is trivial: it just returns velocity and position).

From a tutorial I have written, that only extrapolates position and not velocity, this looks like:
```
    while c < n
        prevpos = p.pos; prevvel = p.vel
        i, ct, cp = bounce!(p, bd)
        xs, ys = extrapolate(p, prevpos, prevvel, ct)
        push!(t, ct)
        append!(xt, xs)
        append!(yt, ys)
        c += 1
    end

extrapolate(p::Particle, prevpos, prevvel, ct) = p.pos

function extrapolate(p::MagneticParticle, prevpos, prevvel, t)
    φ0 = atan(prevvel[2], prevvel[1])
    s0, c0 = sincos(φ0)
    x0 = prevpos[1]; y0 = prevpos[2]
    xt = [x0]; yt = [y0]; ω = p.ω
    tvec = 0.0:0.01:t
    for td in tvec
        s, c = sincos(p.ω*td + φ0)
        push!(xt, s/ω + x0 - s0/ω)
        push!(yt, -c/ω + y0 + c0/ω) #vx0 is cos(φ0)
    end
    return xt, yt
end
```

### PR DESCRIPTION
**Pull request by lhupe**
This PR will eventually fix #162 ~and provide a cool new animation function~. It also fixes a hitherto unnoticed bug with #161, where the `plot` method for particles was accidentaly overwritten with an empty function.
### Still to do
- [ ] Update tests
- [ ] Add ray-splitting to `timeseries`
- [x] Decide whether `timeseries` stuff should move into its own file (pinging @Datseris)

### CODE REVIEW

**Comment by Datseris** [comment]
Can you please contribute the animation in a _separate_ pr?

**Comment by lhupe** [comment]
OK. What is your opinion on putting the `construct` and `extrapolate` stuff in its own file?

**Comment by Datseris** [comment]
Sure you can do this, but `evolve` should go there as well. Name it `timeseries` or somethin.

File: src/timeevolution/timeseries.jl [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+    return poss, vels, tvec
+end
+
+function extrapolate(p::Particle{T}, prevpos::SV{T}, prevvel::SV{T}, ct::T,
+                     dt::T) where {T <: AbstractFloat}
+
+    poss = SV{T}[]
+    vels = SV{T}[]
+
+    tvec = collect(0:dt:ct)
```
Comment (Datseris) [inline src/timeevolution/timeseries.jl]: Its clear from here that the length of `poss` and `vels` etc. is known exactly. It is much better to initialize them with proper length and zeros than to `push!` element by element.

At the end you can push the single final element if need be.

File: src/timeevolution/timeseries.jl [hunk 2]
```diff
@@ -0,0 +1,326 @@
+export evolve!, evolve, construct, extrapolate, timeseries!, timeseries
```
Comment (Datseris) [inline src/timeevolution/timeseries.jl]: maybe `extrapolate` should't be exported? We aren't going to document it anyways, right?

**Comment by lhupe** [comment]
In general, I agree. I'm not so sure about the optional bit, because I think we should still plot the velocity vector on the particle for the animation, which would require timeseries to also return the velocities. But as this is for a future PR anyway, it doesn't matter right now

**Comment by Datseris** [comment]
YEah I corrected the optional bit, I haven't thought it through.

**Comment by lhupe** [comment]
also, this will leave the package without an option to generate timeseries from ray-splitting data, at least if I also remove the ray-splitting `construct`

**Pull request merged**